src/server/middleware/prometheus.ts

```diff
@@ -0,0 +1,54 @@
+import fastify, {
+  FastifyInstance,
+  FastifyReply,
+  FastifyRequest,
+} from "fastify";
+import { env } from "../../utils/env";
+import { enginePromRegister, recordMetrics } from "../../utils/prometheus";
+
+export const withPrometheus = async (server: FastifyInstance) => {
+  if (!env.METRICS_ENABLED) {
+    console.log("Metrics are disabled");
+    return;
+  }
+
+  console.log("Metrics are enabled");
+  server.addHook(
+    "onResponse",
+    (req: FastifyRequest, res: FastifyReply, done) => {
+      const { method } = req;
+      const url = req.routeOptions.url;
+      const { statusCode } = res;
+      const duration = res.elapsedTime; // Use Fastify's built-in timing
+
+      // Record metrics
+      recordMetrics({
+        event: "response_sent",
+        params: {
+          endpoint: url ?? "",
+          statusCode: statusCode.toString(),
+          duration: duration,
+          method: method,
+        },
+      });
+
+      done();
+    },
+  );
+  // Expose metrics endpoint
+
+  const metricsServer = fastify({
+    disableRequestLogging: true,
+  });
+
+  metricsServer.get("/metrics", async (request, reply) => {
+    reply.header("Content-Type", enginePromRegister.contentType);
+    return enginePromRegister.metrics();
+  });
+
+  await metricsServer.ready();
+  metricsServer.listen({
+    host: env.METRICS_HOST ?? env.HOST,
```

Comment:
In what situation should this be overridden? 
From the [`listen()` docs](https://fastify.dev/docs/latest/Reference/Server/#listen) you'd only override it to support IPv4/v6 addresses. I'm not really sure of the behavior we'd want to allow, so I wonder if we should just default to `env.HOST` since we want this metrics server to probably just listen on the same IP types?

tl;dr: Should we remove `METRICS_HOST` as an env var?